Add metadata information to POMs

lineargraph opened the pull request.

lineargraph committed:
Add metadata information to POMs

modmuss50 commented on build.gradle:
```diff
@@ -509,6 +509,23 @@ task runProductionAutoTestServer(type: JavaExec, dependsOn: [remapJar, remapTest
+		}
+	}
+	pom.scm {
+		url = "https://github.com/FabricMC/fabric"
```
A few more things can be added to the scm as seen here: https://github.com/FabricMC/mapping-io/blob/master/build.gradle#L104

modmuss50 commented on build.gradle:
```diff
@@ -509,6 +509,23 @@ task runProductionAutoTestServer(type: JavaExec, dependsOn: [remapJar, remapTest
 	}
 }
 
+def addPomMetadataInformation(MavenPom pom) {
```
Would be nice to also set name and url. description could possibly be driven by the fabric.mod.json file, but thats a little more effort.

modmuss50 replied:
issueManagement might be another useful one to add.

modmuss50 commented on build.gradle:
```diff
@@ -509,6 +509,23 @@ task runProductionAutoTestServer(type: JavaExec, dependsOn: [remapJar, remapTest
+def addPomMetadataInformation(MavenPom pom) {
+	pom.licenses {
+		license {
+			name = "Apache 2.0"
```
```suggestion
			name = "Apache-2.0"
```

According to the [docs](https://maven.apache.org/pom.html#Licenses) Using an [SPDX identifier](https://spdx.org/licenses/) as the license name is recommended.

modmuss50 reviewed (changes requested):
If we are going to do this, I think we should make sure to populate it fully. See my comments and https://github.com/FabricMC/mapping-io/blob/master/build.gradle#L99 for some ideas.

lineargraph committed:
Add name, description, url and issue management

Also update the license name and an url to the fabric website
Pulls name and description from the fabric.mod.json

modmuss50 commented on build.gradle:
```diff
@@ -442,7 +445,8 @@ dependencies {
 
-import net.fabricmc.loom.util.Platform
+
+import org.kohsuke.github.GHReleaseBuilder
```
No need to move these imports, I think its generally better to keep the imports close to where they are used.

modmuss50 commented on build.gradle:
```diff
@@ -509,6 +513,37 @@ task runProductionAutoTestServer(type: JavaExec, dependsOn: [remapJar, remapTest
 }
 
+def addPomMetadataInformation(Project project, MavenPom pom) {
+	def modJsonFile = project.file("src/main/resources/fabric.mod.json")
+	if (!modJsonFile.exists())
+		modJsonFile = project.file("src/client/resources/fabric.mod.json")
+	def modJson = new JsonSlurper().parse(modJsonFile)
+	pom.name = modJson.name
```
Great idea 👍

modmuss50 commented on build.gradle:
```diff
@@ -509,6 +513,37 @@ task runProductionAutoTestServer(type: JavaExec, dependsOn: [remapJar, remapTest
+		modJsonFile = project.file("src/client/resources/fabric.mod.json")
+	def modJson = new JsonSlurper().parse(modJsonFile)
+	pom.name = modJson.name
+	pom.url = "https://github.com/FabricMC/fabric/tree/${grgit.head().abbreviatedId}/${project.rootDir.relativePath(project.projectDir)}"
```
I would just hard code the URL to `https://github.com/FabricMC/fabric`, as this wont work if someone tries to build it outside of a github repo. (Such as when you just download the zip from github)

modmuss50 commented on build.gradle:
```diff
@@ -509,6 +513,37 @@ task runProductionAutoTestServer(type: JavaExec, dependsOn: [remapJar, remapTest
+	pom.licenses {
+		license {
+			name = "Apache-2.0"
+			url = "https://github.com/FabricMC/fabric/blob/${grgit.head().abbreviatedId}/LICENSE"
```
Again, dont use git here.

modmuss50 commented on build.gradle:
```diff
@@ -668,7 +710,7 @@ if (signingEnabled) {
 }
 
-import org.kohsuke.github.GHReleaseBuilder
+
```
Revert these import changes.

modmuss50 reviewed (changes requested):
This is looking good, nice work reading it from the FMJ. Im not too keen on using grgit like this as people can and do build it without being in a git repo.

lineargraph committed:
Fix imports and make git optional

lineargraph commented:
@modmuss50 I now made grgit optional. I can also entirely remove it, if you want that.

modmuss50 commented:
> @modmuss50 I now made grgit optional. I can also entirely remove it, if you want that.

I think it would be best, helps enusre that its consitant between runs.

lineargraph committed:
Remove commit from maven metadata
